stdlib/src/utils/string_slice.mojo

```diff
@@ -640,16 +658,28 @@ struct StringSlice[is_mutable: Bool, //, origin: Origin[is_mutable].type,](
         )
 
     @always_inline
-    fn as_bytes(self) -> Span[Byte, origin]:
-        """Get the sequence of encoded bytes of the underlying string.
+    fn as_bytes[
+        is_mutable: Bool = is_mutable,
+        origin: Origin[is_mutable]
+        .type = _lit_mut_cast[origin, is_mutable]
+        .result,
+    ](self) -> Span[Byte, origin]:
+        """Returns a contiguous slice of bytes.
+
+        Parameters:
+            is_mutable: Whether the result will be mutable.
+            origin: The origin of the data.
 
         Returns:
-            A slice containing the underlying sequence of encoded bytes.
+            A contiguous slice pointing to bytes.
+
+        Notes:
+            This does not include the trailing null terminator.
         """
-        return self._slice
+        return rebind[Span[Byte, origin]](self._slice)
```

Comment:
**Question** Why is the rebind necessary instead over just the span constructor?

Reply:
I don't see why we'd need to build a new instance here. AFAIK rebind is just telling the compiler "trust me it's this type". I'm mostly just trying stuff out, WDYT?

Reply:
Ah, I see.  The origins may be different, which is why the rebind is needed.  Makes sense 👍 

I agree now we don't need an explicit constructor call.